src/inference/src/dev/core_impl.cpp

```diff
@@ -664,22 +664,17 @@ void ov::CoreImpl::set_property(const std::string& device_name, const AnyMap& pr
                     "set_property is supported only for BATCH itself (without devices). "
                     "You can configure the devices with set_property before creating the BATCH on top.");
 
-    bool isMetaDevice = device_name.find("AUTO") != std::string::npos ||
-                        device_name.find("MULTI") != std::string::npos ||
-                        device_name.find("HETERO") != std::string::npos;
-    if (!isMetaDevice) {
-        // unsupport to set ov::device::properties to HW device through this function
-        auto devices = get_registered_devices();
-        for (auto&& config : properties) {
-            auto parsed = parseDeviceNameIntoConfig(config.first);
-            auto is_secondary_config_for_hw_device =
-                std::any_of(devices.begin(), devices.end(), [&](const std::string& device) {
-                    return device == parsed._deviceName;
-                });
-            OPENVINO_ASSERT(!is_secondary_config_for_hw_device,
-                            "set_property only supported ov::device::propreties for Meta device (AUTO/MULTI/HETERO). "
-                            "You can configure the devices through the compile_model()/loadNetwork() API.");
-        }
+    // unsupport to set ov::device::properties to HW device through this function
+    auto devices = get_registered_devices();
+    for (auto&& config : properties) {
+        auto parsed = parseDeviceNameIntoConfig(config.first);
+        auto is_secondary_config_for_hw_device =
+            std::any_of(devices.begin(), devices.end(), [&](const std::string& device) {
+                return device == parsed._deviceName;
+            });
```

Comment:
I don't understand this logic, why not just search `ov::device::properties.name()` among `properties` ?

Reply:
`ov::device::properties` doesn't have this member function `name()`, and the element in `properties` is a std::pair object consisted of property name and property value instead of a property object. Fell free to correct me if I misunderstand something.

Reply:
@yangwang201911 @ilya-lavrenov Actually that's true https://github.com/openvinotoolkit/openvino/blob/9cd23790c7176ecc30ad23694855340766f48020/src/inference/include/openvino/runtime/properties.hpp#L657 But I'm wondering why we can't make it common property with name `DEVICE_PROPERTIES` and map as value? Does anybody know it? I can suppose that problem is in old api, where we have map string to string. And maybe `CPU: NUM_STREAMS 8` can be parsed more easy when device name is key. But I think we have to change it in the future.

Reply:
@yangwang201911 BTW is it working with device with id?

Reply:
it works for device with id checking.